scalalib/overrides-2.12/scala/annotation/nowarn.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+// This file was overidden to add @nowarn annotation removing compile time warnings
```

Comment:
Why do we need this?

Reply:
When compiling in 2.12.13 we would get this warning, treated as an error in our build
```
[error] scala-native/scalalib/target/scalaSources/2.12.13/scala/annotation/nowarn.scala:34:7: Implementation restriction: subclassing Classfile does not
[error] make your annotation visible at runtime.  If that is what
[error] you want, you must write the annotation class in Java.
[error] class nowarn(value: String = "") extends ClassfileAnnotation
[error]       ^
[error] one error found

```
Logic in Scala 2.12.x compiler causing this problem can be seen [here](https://github.com/scala/scala/blob/e5580a6f616eeee0680c3a167d86f4bb755fc951/src/compiler/scala/tools/nsc/typechecker/Typers.scala#L1837)
Definition of `nowarn` is defined [here](https://github.com/scala/scala/blob/2.12.x/src/library/scala/annotation/nowarn.scala)

Actually I've made mistake in this override. I've commited change where `class nowarn`  extends `StaticAnnotation` instead of original `ClassfileAnnotation`, as I thought that for some adding missing usage of `@nowarn` on class definition caused this problems. 

Probably we should conditionally remove treating warnings as errors for 2.12.13 in scalalib project.

Reply:
If it's in the `scalalib`, it's not our problem. We should simply disable fatal warnings in the scalalib rather than chaning the sources just to get rid of warnings. Warnings should be fatal for those writing the code in scala/scala, not for us who are just reusing and recompiling that code.